### PR TITLE
Add support to load lite or full implementations.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opencensus/trace/Tracing.java [hunk 1]
```diff
@@ -82,6 +82,14 @@ static TraceComponent loadTraceComponent(ClassLoader classLoader) {
           Class.forName("io.opencensus.trace.TraceComponentImpl", true, classLoader),
           TraceComponent.class);
     } catch (ClassNotFoundException e) {
+      logger.log(Level.FINE, "Try to load lite implementation.", e);
```
Comment (sebright) [inline api/src/main/java/io/opencensus/trace/Tracing.java]: Should this message give more detail, like "Couldn't load full implementation for TraceComponent, now trying to load lite implementation."?

File: core/src/main/java/com/google/instrumentation/stats/Stats.java [hunk 2]
```diff
@@ -13,34 +13,54 @@
...
+  @VisibleForTesting
+  @Nullable
+  static StatsManager loadStatsManager(ClassLoader classLoader) {
+    try {
+      // Call Class.forName with literal string name of the class to help shading tools.
+      return Provider.createInstance(
+          Class.forName("com.google.instrumentation.stats.StatsManagerImpl", true, classLoader),
+          StatsManager.class);
+    } catch (ClassNotFoundException e) {
+      logger.log(Level.FINE, "Try to load lite implementation.", e);
```
Comment (sebright) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: Same here

File: core/src/test/java/com/google/instrumentation/stats/StatsTest.java [hunk 3]
```diff
@@ -13,17 +13,49 @@
...
+/** Tests for {@link Stats}. */
 @RunWith(JUnit4.class)
 public final class StatsTest {
-  @Test(expected = AssertionError.class)
-  public void testConstructor() {
-    new Stats();
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void loadTraceService_UsesProvidedClassLoader() {
```
Comment (sebright) [inline core/src/test/java/com/google/instrumentation/stats/StatsTest.java]: The first two method names are incorrect.

File: api/src/main/java/io/opencensus/trace/Tracing.java (hunk 1 above)
Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/trace/Tracing.java]: Done.

File: core/src/main/java/com/google/instrumentation/stats/Stats.java (hunk 2 above)
Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: Done.

File: core/src/test/java/com/google/instrumentation/stats/StatsTest.java (hunk 3 above)
Comment (bogdandrutu) [inline core/src/test/java/com/google/instrumentation/stats/StatsTest.java]: Done.

**Pull request merged**